perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java

```diff
@@ -29,6 +40,147 @@ public void setUp() throws Exception {
+
+		perun.getConsentsManagerBl().createConsent(sess, consent1);
+		perun.getConsentsManagerBl().createConsent(sess, consent2);
+
+		assertEquals(2, consentsManagerEntry.getAllConsents(sess).size());
+	}
+
+	@Test
+	public void getConsentsForUser() throws Exception {
+		System.out.println(CLASS_NAME + "getConsentsForUser");
+
+		User user = setUpUser("John", "Doe");
+
+		Facility facility1 = setUpFacility();
+		Facility facility2 = new Facility();
+		facility2.setName("TestFacility2");
+
+		// createFacility method creates also new Consent Hub
+		perun.getFacilitiesManager().createFacility(sess, facility2);
+
+		Consent consent1 = new Consent(-1, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility1.getName()), new ArrayList<>());
+		Consent consent2 = new Consent(-11, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility2.getName()), new ArrayList<>());
+
+		perun.getConsentsManagerBl().createConsent(sess, consent1);
+		perun.getConsentsManagerBl().createConsent(sess, consent2);
+
+
+		assertEquals(2, consentsManagerEntry.getConsentsForUser(sess, user.getId()).size());
+		assertEquals(2, consentsManagerEntry.getConsentsForUser(sess, user.getId(), ConsentStatus.UNSIGNED).size());
+		assertEquals(0, consentsManagerEntry.getConsentsForUser(sess, user.getId(), ConsentStatus.GRANTED).size());
+	}
+
+	@Test
+	public void getConsentsForConsentHub() throws Exception {
+		System.out.println(CLASS_NAME + "getConsentsForConsentHub");
+
+		Facility facility = setUpFacility();
+
+		User user1 = setUpUser("John", "Doe");
+		User user2 = setUpUser("Donald", "Trump");
+
+		Consent consent1 = new Consent(-1, user1.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+		Consent consent2 = new Consent(-11, user2.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+
+		perun.getConsentsManagerBl().createConsent(sess, consent1);
+		perun.getConsentsManagerBl().createConsent(sess, consent2);
+
+		assertEquals(2, consentsManagerEntry.getConsentsForConsentHub(sess, perun.getConsentsManager().getConsentHubByName(sess, facility.getName()).getId()).size());
+		assertEquals(2, consentsManagerEntry.getConsentsForConsentHub(sess, perun.getConsentsManager().getConsentHubByName(sess, facility.getName()).getId(), ConsentStatus.UNSIGNED).size());
+		assertEquals(0, consentsManagerEntry.getConsentsForConsentHub(sess, perun.getConsentsManager().getConsentHubByName(sess, facility.getName()).getId(), ConsentStatus.GRANTED).size());
+	}
+
+	@Test
+	public void getConsentById() throws Exception {
+		System.out.println(CLASS_NAME + "getConsentById");
+
+		Facility facility = setUpFacility();
+		User user = setUpUser("John", "Doe");
+
+		AttributeDefinition attrDef = new AttributeDefinition();
+		attrDef.setNamespace(AttributesManager.NS_FACILITY_ATTR_DEF);
```

Comment:
Consents will be related to user/member attributes, it would be better to use it that way in this test also. It would be nice to  have a check that the correct facility is associated with the returned consentHub. Also a test for `getConsentsForUserAndConsentHub` would be nice to have.